Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp

```diff
@@ -101,27 +69,164 @@ RemoteMediaSessionManager::~RemoteMediaSessionManager()
 }
 
 void RemoteMediaSessionManager::setCurrentSession(WebCore::PlatformMediaSessionInterface& session)
 {
     ALWAYS_LOG(LOGIDENTIFIER, session.logIdentifier(), ", size = ", sessions().computeSize());
     PlatformMediaSessionManager::setCurrentSession(session);
-    send(Messages::RemoteMediaSessionManagerProxy::SetCurrentSession(platformSessionState(session)));
+
+    send(Messages::RemoteMediaSessionManagerProxy::SetCurrentMediaSession(currentSessionState(session)));
 }
 
 void RemoteMediaSessionManager::updateSessionState()
 {
-    send(Messages::RemoteMediaSessionManagerProxy::UpdateSessionState());
+    send(Messages::RemoteMediaSessionManagerProxy::UpdateMediaSessionState());
+}
+
+RefPtr<WebCore::PlatformMediaSessionInterface> RemoteMediaSessionManager::sessionWithIdentifier(WebCore::MediaSessionIdentifier identifier)
+{
+    return firstSessionMatching([identifier](auto& session) {
+        return session.mediaSessionIdentifier() == identifier;
+    }).get();
+}
+
+void RemoteMediaSessionManager::clientShouldResumeAutoplaying(WebCore::MediaSessionIdentifier identifier)
+{
+    if (RefPtr session = sessionWithIdentifier(identifier))
+        session->client().resumeAutoplaying();
+}
+
+void RemoteMediaSessionManager::clientMayResumePlayback(WebCore::MediaSessionIdentifier identifier, bool shouldResume)
+{
+    if (RefPtr session = sessionWithIdentifier(identifier))
+        session->client().mayResumePlayback(shouldResume);
+}
+
+void RemoteMediaSessionManager::clientShouldSuspendPlayback(WebCore::MediaSessionIdentifier identifier)
+{
+    if (RefPtr session = sessionWithIdentifier(identifier))
+        session->client().suspendPlayback();
 }
 
+void RemoteMediaSessionManager::clientSetShouldPlayToPlaybackTarget(WebCore::MediaSessionIdentifier identifier, bool shouldPlay)
+{
+    if (RefPtr session = sessionWithIdentifier(identifier))
+        session->client().setShouldPlayToPlaybackTarget(shouldPlay);
+}
+
+void RemoteMediaSessionManager::clientDidReceiveRemoteControlCommand(WebCore::MediaSessionIdentifier identifier, WebCore::PlatformMediaSessionRemoteControlCommandType command, WebCore::PlatformMediaSessionRemoteCommandArgument argument)
+{
+    if (RefPtr session = sessionWithIdentifier(identifier))
+        session->client().didReceiveRemoteControlCommand(command, argument);
+}
+
+RemoteMediaSessionState& RemoteMediaSessionManager::currentSessionState(const WebCore::PlatformMediaSessionInterface& session)
+{
+    auto addResult = m_cachedSessionState.ensure(session.mediaSessionIdentifier(), [&] {
+        return makeUniqueRef<RemoteMediaSessionState>(fullSessionState(session));
+    });
+
+    if (!addResult.isNewEntry)
+        updateCachedSessionState(session, addResult.iterator->value);
+
+    return addResult.iterator->value;
+}
+
+void RemoteMediaSessionManager::updateCachedSessionState(const WebCore::PlatformMediaSessionInterface& session, RemoteMediaSessionState& state)
+{
+    state.state = session.state();
```

Comment:
Prefers the previously used syntax and it's less verbose.

```suggestion
    state = {
        .state = session.state(),
        .stateToRestore = session.stateToRestore(),
```
....

Reply:
I don't think it can be less verbose, it is initializing a `RemoteMediaSessionState` from a `PlatformMediaSessionInterface`